### ISSUE TITLE
HTTP status for missing tile

### ISSUE THREAD

**Issue by MatteoColombo** [CartoDB/mobile-sdk#402]
Hi, I'm using two HTTPTileDataSource layers that have data only for some tiles. 
If there is no data for a tile server returns 404, but I noticed that in this way the tile loading if some are missing is quite slow and requests are repeated several times.
Which is the correct status code to make sure that request is performed just once?

**Comment by mtehver** [comment]
Hi. Correct, SDK does not keep track of such tiles and sends multiple requests to HTTPTileDataSource. The use case is somewhat untypical and there is a workaround for this - make your own data source class by subclassing HTTPTileDataSource. Simply override 'loadTile' method and keep a list of bad tiles. If a new request for a tile comes in for a tile already in the list, simply return null. Otherwise send the request to original 'loadTile' implementation.

**Comment by MatteoColombo** [comment]
Thanks! Instead of keeping a list of bad tiles I return a TileData with empty content and it's working fine. 
The only thing that I noticed is that if I set the MaxAge of the TileData, when the tile expires it's not refreshed with new data on the screen unless I move the map and come back. Is this the intended behavior? Is there any way to automatically make the tile layer call 'loadTile' when a tile expires?

**Comment by mtehver** [comment]
That's correct, SDK currently does not automatically refresh the map when tiles expire. If you are using custom datasource, you can force this behavior from the datasource side by using a timer and calling `notifyTilesChanged` when the tiles have expired.
